staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2263,31 +2263,31 @@ const (
 	PullIfNotPresent PullPolicy = "IfNotPresent"
 )
 
-// ResourceResizePolicy specifies how Kubernetes should handle resource resize.
-type ResourceResizePolicy string
+// ResourceResizeRestartPolicy specifies how to handle container resource resize.
+type ResourceResizeRestartPolicy string
 
-// These are the valid resource resize policy values:
+// These are the valid resource resize restart policy values:
 const (
-	// RestartNotRequired tells Kubernetes to resize the container in-place
+	// 'RestartNotRequired' means Kubernetes will try to resize the container
 	// without restarting it, if possible. Kubernetes may however choose to
 	// restart the container if it is unable to actuate resize without a
 	// restart. For e.g. the runtime doesn't support restart-free resizing.
-	RestartNotRequired ResourceResizePolicy = "RestartNotRequired"
-	// 'RestartRequired' tells Kubernetes to resize the container in-place
+	RestartNotRequired ResourceResizeRestartPolicy = "RestartNotRequired"
```

Comment:
`restartPolicy: RestartNotRequired`  vs.  `restartPolicy: NotRequired`  - I prefer "NotRequired" (like we say `restartPolicy: Always` rather than `RestartAlways`).

It's less obvious what "RestartContainer" should be.  `restartPolicy: Container` ?  or "KillContainer"?  Or "Required" ?  Do we think there could ever conceivably be a "RestartPod" mode?  That seems like a stretch.  Can you think of any other plausible mode to think about names?

Reply:
should this be `PreferNoRestart`? So later if we will need the policy indicating that the change SHALL NOT be applied if restart is required we can add it?

Reply:
I had to go find what I said before and, for once, I agree with past-me.

Context: https://github.com/kubernetes/kubernetes/pull/111946#discussion_r1068797501 wherein I wasn't as clear as I could be.

I was thinking `restartPolicy: (NotRequired | RestartContainer)` was OK.  It's a good question whether "Not Required" means "not required" or "require not".  We could call that latter "Forbidden" or something?

Reply:
Forbidden works for the future use. There are use cases in gaming that will benefit from `Forbidden` so I want to make  sure it fits into the API extensibility

Reply:
Good call out with gaming.  Would they actually try an in-place resize?

Reply:
we definitely will help to try it.

Reply:
@thockin Yes `RestartPod` was discussed as a [plausible use case](https://github.com/kubernetes/enhancements/pull/686#discussion_r326999056) where the workload might require re-running init containers. This is why I feel `RestartContainer` is the right choice. The name `RestartNotRequired` comes from allowing user to specify their app can be resized without restarting but we make no guarantees of app won't be restarted (e.g. the case where runtime does not support restart-free resizing).

Reply:
Should "container" be included in the name (like `RestartContainer`) below? Thinking about the pod restart usage scenarios mentioned above.

Should the word require/prefer be part of the name, like `PreferNoRestart`, `RequireRestartContainer`?

Reply:
I can buy the stutter in this case.   I do find `restartPolicy: Container` or `restartPolicy: Pod` to be awkward, so `restartPolicy: (NotRequired | RestartContainer)` is not so terribly offensive to mine eyes, with "RestartPod" or "Forbidden" being plausible extensions. 

I'd still go with "NotRequired" instead of "RestartNotRequired" - the stutter doesn't add any clarity there.